apis/v1beta1/providerconfig_types.go

```diff
@@ -17,32 +17,38 @@ limitations under the License.
 package v1beta1
 
 import (
-	"github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
 )
 
 // A ProviderConfigSpec defines the desired state of a ProviderConfig.
 type ProviderConfigSpec struct {
-	// CredentialsSecretRef references a specific secret's key that contains
-	// the credentials that are used to connect to the GCP API.
-	CredentialsSecretRef v1alpha1.SecretKeySelector `json:"credentialsSecretRef"`
+	v1alpha1.ProviderConfigSpec `json:",inline"`
```

Comment:
Do we have to do that? I recently changed this line in order to make `credentialsSecretRef` required because it is required right now for this provider.

Reply:
I think there's value in embedding this type. It helps us keep the `credentialsSecretRef` field including its docstring and JSON tag consistent across providers, and would do the same for any other spec fields we wanted most providers to support.

That said, I agree that `credentialsSecretRef` should be required by default. I believe AWS is the only provider where it's actually optional. My preference would be to update the embedded type to make the field required, and have AWS opt out of using the embedded type. I'll handle that in a follow on PR since it will affect all providers.